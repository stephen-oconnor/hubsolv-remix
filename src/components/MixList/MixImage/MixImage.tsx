import React from 'react';
import PropTypes from 'prop-types';
import s from './styles.module.scss';

interface MixImageProps {
	src: string;
	altTag: string;
}

const MixImage: React.FC<MixImageProps> = ({ src, altTag }) => {
	return (
		<div className={s.cover}>
			<img src={src} alt={altTag} />
		</div>
	);
};

MixImage.propTypes = {
	src: PropTypes.string.isRequired,
	altTag: PropTypes.string.isRequired,
};

export default MixImage;
