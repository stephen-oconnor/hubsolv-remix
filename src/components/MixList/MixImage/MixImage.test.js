import React from 'react';
import MixImage from './MixImage';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<MixImage />', () => {
	it('matches the snapshot', () => {
		const tree = shallow(<MixImage src={'test-path.jpg'} alt={'Something meaningful'} />);
		expect(toJson(tree)).toMatchSnapshot();
	});

	it('should contain 1 img element', () => {
		const wrapper = shallow(<MixImage src={'test-path.jpg'} alt={'Something meaningful'} />);
		expect(wrapper.find('img').length).toBe(1);
	});

	it('img element should have a src and alt', () => {
		const wrapper = shallow(<MixImage src={'test-path.jpg'} alt={'Something meaningful'} />);

		expect(wrapper.find({ src: 'test-path.jpg' }).length).toBe(1);
		expect(wrapper.find({ alt: 'Something meaningful' }).length).toBe(1);
	});
});
