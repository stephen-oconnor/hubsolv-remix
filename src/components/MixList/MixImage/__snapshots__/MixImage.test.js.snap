// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MixImage /> matches the snapshot 1`] = `
<img
  alt="Something meaningful"
  className="image"
  src="test-path.jpg"
/>
`;
